LibreNMS/Validations/Python.php

```diff
@@ -0,0 +1,143 @@
+<?php
+/**
+ * Python.php
+ *
+ * Check that various Python modules and functions exist.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * @package    LibreNMS
+ * @link       http://librenms.org
+ * @copyright  2020 Thomas Berberich
+ * @author     Thomas Berberich <sourcehhdoctor@gmail.com>
+ */
+
+namespace LibreNMS\Validations;
+
+use LibreNMS\Config;
+use LibreNMS\ValidationResult;
+use LibreNMS\Validator;
+
+class Python extends BaseValidation
+{
+    const PYTHON_MIN_VERSION = '3.4.0';
+    const PYTHON_RECOMMENDED_VERSION = '3.5.2';
+
+    const PYTHON_REQUIREMENTS_FILE = 'requirements.txt';
+
+    public static function pythonVersion()
+    {
+        return explode(' ', exec('python3 --version'))[1];
+    }
+
+    /**
+     * Validate this module.
+     * To return ValidationResults, call ok, warn, fail, or result methods on the $validator
+     *
+     * @param Validator $validator
+     */
+    public function validate(Validator $validator)
+    {
+        $this->checkVersion($validator);
+        $this->checkExtensions($validator);
```

Comment:
FYI, there is already as script that checks this.  Although it does not give detailed output.
`scripts/check_requirements.py`

Reply:
oh nice. I'll take a look on it

Reply:
enhanced `scripts/check_requirements.py` a little bit ...